src/Microsoft.IdentityModel.Tokens/AsymmetricSignatureProvider.cs

```diff
@@ -146,6 +146,9 @@ public AsymmetricSignatureProvider(SecurityKey key, string algorithm, bool willC
             if (asymmetricAlgorithmResolver != null)
             {
                 AsymmetricAlgorithm asymmetricAlgorithm = asymmetricAlgorithmResolver(key, algorithm, willCreateSignatures);
+                if (asymmetricAlgorithm == null)
+                    throw LogHelper.LogException<ArgumentOutOfRangeException>(LogMessages.IDX10646, nameof(key), algorithm);
```

Comment:
ArgumentOutOfRangeException is fine. The error message text will help out devs.
On second thought, just pass 'key' instead of 'key.ToString()'. That is our standard.